diff --git a/lib/typescript.js b/lib/typescript.js
--- a/lib/typescript.js
+++ b/lib/typescript.js
@@ -3766,7 +3766,7 @@
   }
 }
 function isNodeLikeSystem() {
-  return typeof process !== "undefined" && !!process.nextTick && !process.browser && typeof require !== "undefined";
+  return typeof WebSocketPair === "undefined" && typeof process !== "undefined" && !!process.nextTick && !process.browser && typeof require !== "undefined";
 }
 
 // src/compiler/debug.ts
